src/libraries/System.Text.Json/docs/collection_support.md

```diff
@@ -0,0 +1,129 @@
+# JsonSerializer collection support in 3.1.x
+
+All collections (any type that derives from IEnumerable) are supported for
+serialization with JsonSerializer; we simply call the .GetEnumerator() method
+of the collection, and write the elements if they are supported.
+Deserialization is more complicated and is not supported for all collections.
+
+This document gives a quick overview of which collections are supported.
+
+## Built-in types
+
+### System.Collections
+
+| Type | Serialization | Deserialization |
+| --- | --- | --- |
+| ArrayList | Supported | Supported |
+| BitArray | Supported | Not supported |
+| Hashtable | Supported | Supported |
+| Queue | Supported | Supported |
+| SortedList | Supported | Supported |
+| Stack | Supported | Supported |
+| DictionaryEntry | Supported | Supported |
+| ICollection | Supported | Supported |
+| IDictionary | Supported | Supported |
+| IEnumerable | Supported | Supported |
+| IList | Supported | Supported |
+
+### System.Collections.Generic
+
+| Type | Serialization | Deserialization |
+| --- | --- | --- |
+| Dictionary<string, TValue> | Supported | Supported |
+| HashSet<T> | Supported | Supported |
+| LinkedList<T> | Supported | Supported |
+| LinkedListNode<T> | Supported | Not supported |
+| List<T> | Supported | Supported |
+| Queue<T> | Supported | Supported |
+| SortedDictionary<string,TValue> | Supported | Supported |
+| SortedList<string,TValue> | Supported | Supported |
+| SortedSet<T> | Supported | Supported |
+| Stack<T> | Supported | Supported |
```

Comment:
Should we put an asterisk next to those where the behavior might be unexpected (so we can add notes about them below the table)? For example, serializing `Stack<T>` types.